cloud/scope/machine.go

```diff
@@ -327,3 +340,30 @@ func (m *MachineScope) GetBootstrapData() (string, error) {
 	}
 	return string(value), nil
 }
+
+func fetchKeyID(key *infrav1beta1.IBMVPCResourceReference, m *MachineScope) (*string, error) {
+	if key.ID != nil {
+		return key.ID, nil
+	}
+
+	if key.Name != nil {
+		keys, _, err := m.IBMVPCClient.ListKeys(&vpcv1.ListKeysOptions{})
+		if err != nil {
+			m.Logger.Error(err, "Failed to get keys")
+			return nil, err
+		}
+
+		for _, k := range keys.Keys {
+			if *k.Name == *key.Name {
+				m.Logger.Info("Key found with ID", "Key", *k.Name, "ID", *k.ID)
```

Comment:
this seems like a debug log and shouldn't be enabled by default.

Reply:
Fixed.